### PR TITLE
feat(elasticsearch sink): Add suppress_type_name to allow for ESv8 compatibility

### LINKED ISSUE #9970: Allow supressing `doc_type` setting for elasticsearch sink
As of ElasticSearch 7+, specifying a doc_type is no longer supported. It is deprecated with ES7 (and returns a warning), and not supported at all in ES 8+
https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html

Vector currently has a default of `_doc` for `doc_type` if none is specified. To maintain backwards compatibility, a new config option should be added such as `suppress_doc_type`, which prevent a `doc_type` from being specified in requests.

### PR DESCRIPTION
**Pull request by spencergilbert**
Signed-off-by: Spencer Gilbert <spencer.gilbert@datadoghq.com>

Closes #9970

### CODE REVIEW

File: src/sinks/elasticsearch/encoder.rs [hunk 1]
```diff
@@ -70,18 +72,29 @@ fn write_bulk_action(
...
     doc_type: &str,
+    suppress_type: bool,
     id: &Option<String>,
 ) -> std::io::Result<usize> {
     as_tracked_write(
         writer,
-        (bulk_action, index, doc_type, id),
-        |writer, (bulk_action, index, doc_type, id)| {
+        (bulk_action, index, doc_type, id, suppress_type),
+        |writer, (bulk_action, index, doc_type, id, suppress_type)| {
```
Comment (jszwedko) [inline src/sinks/elasticsearch/encoder.rs:81]: Nit: I personally find this sort of boolean table more readable as a match statement:

```
match (id, supress_type) {
  Some(id), true => {},
  Some(id), false => {}
  None, true => {}
  None, false => {}
}
```

But not a big deal.

**Review by jszwedko — APPROVED** [review]
👍 it might be worth adding a test for `write_bulk_action`.

**Comment by spencergilbert** [comment]
> 👍 it might be worth adding a test for `write_bulk_action`.

Integration test you mean?

File: src/sinks/elasticsearch/encoder.rs (hunk 1 above)
Comment (spencergilbert) [reply src/sinks/elasticsearch/encoder.rs:81]: I also find it super unreadable 👍

**Comment by jszwedko** [comment]
> > 👍 it might be worth adding a test for `write_bulk_action`.
> 
> Integration test you mean?

Integration test would be great. I was just thinking of a unit test for that function though that asserts the encoded JSON is correct given the different parameters.

**Review by fuchsnj — APPROVED** [review]

**Review by StephenWakely — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Looks great! Thanks for adding the tests.

**Review by StephenWakely — APPROVED** [review]
Nice.

**Pull request merged**